packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -867,19 +867,11 @@ export default function(
       }
     }
 
-    let derivedStateFromProps;
```

Comment:
Why did you change the order of `getDerivedStateFromCatch` and `getDerivedStateFromProps`? The new order seems correct, just curious what led you to make the change in this diff.

It looks like the order in the "resume mount" path is different now, though. Should be consistent regardless.

Reply:
> Why did you change the order of `getDerivedStateFromCatch` and `getDerivedStateFromProps`?

This was incidental. I changed the order of `processUpdateQueue` and `getDerivedStateFromProps`. I did that because it seemed easier to get the correct partial state value that way than trying to mess with the update queue and it's memoized vs base state.

> It looks like the order in the "resume mount" path is different now, though. Should be consistent regardless.

Fair 👍

Reply:
One more thing: can you add a test so the order doesn't incidentally change in the future?